### DISCUSSION TITLE
Webserver gunicorn worker keep sliently exiting, help!!

### DISCUSSION THREAD

**Discussion by ORuteMa** [apache/airflow#19925] [Q&A]
When I launch airflow webserver, my webserver gunicorn worker never get ready.
![image](https://user-images.githubusercontent.com/30034544/144198655-e44c6ebf-840a-4028-a0e9-21a49bc11909.png)
Every time I execute ps command, I will get different pid of workers.
![image](https://user-images.githubusercontent.com/30034544/144198575-f493126c-28b7-455c-818e-34f43ca7698e.png)
And the CPU usage keep 100%.
I have no idea whats going on here, ask for some kind help.

--------------------------------------------------------------------

Version=2.2.0, here's the log of webserver.log

____ |( )_______ / /________ __
____ /| |_ /__ / / __ / __ _ | /| / /
___ ___ | / _ / _ / _ / / // / |/ |/ /
// |// // // // _/____/|__/
[2021-12-01 16:20:10,533] {dagbag.py:500} INFO - Filling up the DagBag from /dev/null
[2021-12-01 16:20:10,549] {filesystemcache.py:224} ERROR - set key '__wz_cache_count' -> [Errno 1] Operation not permitted: '/tmp/tmp3l_4ec3d.__wz_cache' -> '/tmp/2029240f6d1128be89ddc32729463129'
[2021-12-01 16:20:10,916] {manager.py:512} WARNING - Refused to delete permission view, assoc with role exists DAG Runs.can_create Admin

**Comment by potiuk** [comment]
Maybe you have problems with permission of the "/tmp" directory. Gunicorn worker by default ? You can try to override that and point to a different directory where you have write access  by defining the environment variable like in this PR (this was fix for CI , but it should work the same in your case)  https://github.com/apache/airflow/pull/9378/files

**Reply by potiuk** [reply]
However this might be rather  Flask cache  (looking at the eror message) so you might want to manually apply this https://github.com/apache/airflow/pull/19208 and make sure your TMPDIR variable is set properly, until we release Airlfow 2.3.0 (the fix will only be released in 2.3.0).

**Reply by potiuk** [reply]
Though maybe even we manage it in 2.2.3 - let's see. I marked it for 2.2.3, seems simple to cherry-pick

**Reply by ORuteMa** [reply]
Sorry for my late reply, thank you very much. It result from the lack of the size of my image. Many repo cant be installed correctly.

**Reply by QuangTruong-at** [reply]
@ORuteMa would you please detail how you fix your problem? We encounter the same issue here, the gunicorn worker processes keep refreshing

**Reply by potiuk** [reply]
FYI: gunicorn workers WILL restart themselves if they are idle for the time specified in https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#web-server-worker-timeout - so continously restarting (default every 2 minutes)  gunicorn workers - if no traffic occurs is perfectly normal behaviour.,

**Reply by ORuteMa** [reply]
> @ORuteMa would you please detail how you fix your problem? We encounter the same issue here, the gunicorn worker processes keep refreshing

Our organization has a pipeline in place to generate corresponding images. The reason behind it was that I mistakenly set the size of the image, which caused Airflow to run into capacity limitations during the virtualenv installation. As a result, many dependencies were missing. However, after adjusting the image size and regenerating it, everything started working properly. I'm not sure if this information is helpful to you.

**Reply by QuangTruong-at** [reply]
Thank you, @ORuteMa ! My case is using ECS Fargate container so it should not be case but I think I found a workaround for this issue with the link from @potiuk ! 
Thanks for the info @potiuk , as our webser will not be used for any processing so I change the config AIRFLOW__WEBSERVER__WORKER_REFRESH_BATCH_SIZE to 0 for not to refresh the worker process then they are not refreshing now. FYI, I got the issue the worker processes keep refreshing within a minute (just 10-20s I think), I think maybe because we don't use them so some settings make the refresh frequently.
